### PR TITLE
parameterize jdk and nginx base images

### PR DESCRIPTION
**Pull request by git-phu**
currently on m1 macs, switching between building for arm vs amd64
architectures is a bit cumbersome because some of the base docker images
have not been parameterized yet, so you will run into build errors unless
you untag those base images every time you switch between architectures.

This PR should allow you switch freely between the two without needing 
that manual step.

This PR also adds a single env var `BUILD_ARCH` that can be used to
switch between building for arm vs amd64.

With this PR we can build and push images for individual platform components
which is much faster than trying to redeploy everything when iterating on
changes that are limited to only a few components.

Ideally we'd have a github action that allowed us to deploy individual platform
components, but until that exists this seems like a reasonable solution for faster
iteration.

### CODE REVIEW

File: docs/contributing-to-airbyte/developing-locally.md
Comment (jrhizor) [inline docs/contributing-to-airbyte/developing-locally.md:37]: FYI, the way this is specified will change with https://github.com/airbytehq/airbyte/pull/11450

Comment (git-phu) [reply docs/contributing-to-airbyte/developing-locally.md:37]: awesome! that change looks great

btw, @jrhizor have you ever actually run into the issue I mentioned in this PR's description?
Basically if you build for amd64, you end up with the amd64 versions of the `openjdk` and `nginx` images in your local docker cache, so if you try to build for arm64 next, you get build errors unless you untag those base images.

Comment (jrhizor) [reply docs/contributing-to-airbyte/developing-locally.md:37]: I haven't actually run into that for some reason. I already have the arm64 version of `nginx:1.19-alpine` cached locally, but I'm not sure why.

Comment (git-phu) [reply docs/contributing-to-airbyte/developing-locally.md:37]: Cool, thanks for the feedback.  I actually haven't found anyone else that has run into this issue yet which is partially why I left this as a draft PR, but it is very possible that my own setup just has something quirky going on which is why I run into this problem.

**Comment by davinchia** [comment]
@git-phu can we wait for @a-honcharenko to merge in his changes before we merge this in?

**Comment by git-phu** [comment]
@davinchia  yup, this is not urgent, just creating a PR to document what I did in case we think it's useful.

**Review by terencecho — APPROVED** [review]
Lgtm

**Review by timroes — APPROVED** [review]
CODEOWNER files looking good to me

**Review by supertopher — APPROVED** [review]

**Comment by git-phu** [comment]
@davinchia looks like this should be good to merge now

**Comment by git-phu** [comment]
yup, just lmk know when is fine

**Comment by davinchia** [comment]
Looks good locally with the multi-arch changes.

**Pull request merged**
